src/components/DataLayersCard.vue

```diff
@@ -174,15 +186,15 @@
           this.$router.push({ path, params })
         } else {
           this.$router.push('/data')
-        }
-
-        //find the layer of the dataset to load on the map based on the chosen values only if the dataset is visible
+        }        
         if (!dataset.visible) {
           this.clearActiveDatasetIds()
+          //TODO: check if there is a better way. I need the layerId to check if the layer is in the activeMapboxLayers
```

Comment:
as in using the params from the router?? (we do this in a few other places in the dgds-front-end code as well)
this is being done also in a few places in the coclico code I see, maybe make a central action in the store or something?